static/js/settings_profile_fields.js

```diff
@@ -98,6 +98,9 @@ exports.reset = function () {
 };
 
 exports.populate_profile_fields = function (profile_fields_data) {
+    $("#account-settings .custom-profile-fields-form").html("");
+    settings_account.add_custom_profile_fields_to_settings();
+
```

Comment:
As we are passing the new list of all custom fields from backend event, it's hard to trace which field is getting updated/created/removed, so we render templates with all new fields. Here templates, represent template for list of custom fields in user settings and template for list of custom fields in organization settings. 
I think we should change this to pass the new changed field only from backend on event, and according to operation just add/remove/update frontend elements of only updated field. Suggestions?

Reply:
That would probably be better from an API perspective.  I think we only do things the current way because we were in a hurry to get something working, and that was easier to make work.  

That said, we might just want to rerender the whole component here anyway, because it's a small component, and as we add things like the 'Hint' feature or a specified, we'll end up with various cases where doing a full rerender is a lot easier to code.